### PR TITLE
[release-1.23] feat: use azcopy for volume cloning

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

[cherry-pick release-1.23] feat: use azcopy for volume cloning
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: .trivyignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+CVE-2023-39325
```
Comment (andyzhangx) [inline .trivyignore:1]: pls patch with https://github.com/kubernetes-sigs/blob-csi-driver/pull/1098, so we don't need this file any more.

Comment (umagnus) [reply .trivyignore]: done. combined #1098 in this pr.

**Pull request merged**